include/arch/arc/v2/irq.h

```diff
@@ -35,6 +35,8 @@ extern void z_arc_firq_stack_set(void);
 extern void arch_irq_enable(unsigned int irq);
 extern void arch_irq_disable(unsigned int irq);
 extern int arch_irq_is_enabled(unsigned int irq);
+extern void sys_trace_isr_enter(void);
```

Comment:
should't these be guarded with the corresponding ifdef?

Reply:
they can be, sure.